recipes/corrade/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+        "fPIC": True,
+        "build_deprecated": True,
+        "with_interconnect": True,
+        "with_main": True,
+        "with_pluginmanager": True,
+        "with_testsuite": True,
+        "with_utility": True,
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version.value) < 14:
+            raise ConanInvalidConfiguration("Corrade requires Visual Studio version 14 or greater")
+        if tools.cross_building(self.settings):
+             raise ConanInvalidConfiguration("This Corrade recipe doesn't support cross building yet")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_STATIC"] = not self.options.shared
+        cmake.definitions["BUILD_DEPRECARED"] = self.options["build_deprecated"]
+        cmake.definitions["WITH_INTERCONNECT"] = self.options["with_interconnect"]
+        cmake.definitions["WITH_MAIN"] = self.options["with_main"]
+        cmake.definitions["WITH_PLUGINMANAGER"] = self.options["with_pluginmanager"] 
+        cmake.definitions["WITH_TESTSUITE"] = self.options["with_testsuite"]
+        cmake.definitions["WITH_UTILITY"] = self.options["with_utility"]
+
+        # TODO: To enable cross-building this executable should probably be outsourced to a separate package corrade-rc
+        cmake.definitions["WITH_RC"] = "OFF"  
+
+        # Corrade uses suffix on the resulting 'lib'-folder when running cmake.install()
+        # Set it explicitly to empty, else Corrade might set it implicitly (eg. to "64")
+        cmake.definitions["LIB_SUFFIX"] = ""
+
+        if self.settings.compiler == "Visual Studio":
+            cmake.definitions["MSVC2015_COMPATIBILITY"] = "ON" if tools.Version(self.settings.compiler.version.value) == 14 else "OFF"
+            cmake.definitions["MSVC2017_COMPATIBILITY"] = "ON" if tools.Version(self.settings.compiler.version.value) == 14 else "OFF"
+
+        cmake.configure(build_folder=self._build_subfolder)
+
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        share_cmake = os.path.join(self.package_folder, 'share', 'cmake', 'Corrade')
+        self.copy('UseCorrade.cmake', src=share_cmake, dst=os.path.join(self.package_folder, 'lib', 'cmake', 'Corrade'))
+        tools.rmdir(os.path.join(self.package_folder, 'share'))
```

Comment:
I see `FindCorrade.cmake` being removed -- why is that? The `find_package(Corrade ...)` call, which calls out into this module, is an essential part of project setup -- checking compiler version, exposing various `CORRADE_TARGET_*` variables to CMake, adding helper macros ...

Or is it being called in a different way and I'm just not aware of it?

Reply:
Okay, so how are you going to ensure the `CORRADE_TARGET_*` variables etc. are exposed to people who *do* use CMake? Because a large portion of tutorials and example code uses them, and those are pretty valuable in distinguishing what platform the user is on, what are the platform properties and what features of the library are enabled or not.

I don't want to have additional maintenance burden by having to keep some `FindCorradeButSlightlyDifferent.cmake` up-to-date just to make Conan happy.